.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches: [master, staging]
+  pull_request:
+    branches: [master, staging]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CHROME_BIN: /usr/bin/google-chrome
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  test:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
+      - name: Install Dependenices
+        run: npm install
+      - name: Run Test with CodeCov
+        env:
+          CHROME_BIN: /usr/bin/google-chrome
+        run: npm run test:codecov
+      - name: Run Lint & Prettier Tests
+        run: |
+          npm run lint:type
+          npm run prettier:check
+      - name: Build staging
+        run: npm run build:stage
+      - name: build production
+        run: npm run build:prod
+      - name: Code coverage test
+        if: success()
+        uses: codecov/codecov-action@v1.1.1
+
+  deployStaging:
```

Comment:
I'm not a big fan of defining multiple `deploy` jobs, but I do get the reason you are doing it because of the different npm tasks that have to be run.

is it possible to use the the environment in the build command? something like this?
```
- name: Build
  run: npm run build:${{ environment }}
```
I'm not sure if I have the right syntax for including the environment here

Reply:
I think you're right; it's best to create a dynamic deployment job for both. 
I don't think it's possible to grab the environment, but we can grab an env var we declare.

Reply:
I'm pretty sure you can get the environment name via one of the contexts they expose in GHA, I'll try and take a look to see if I can find it

Reply:
not finding anything with some good searches, so creating the environment variable will work fine

Reply:
You have to set the environment name on the job, then it understand which preset GitHub environment to use. So I don't believe there's a way to fetch it in GHA, as we define it.